library/core/src/macros/mod.rs

```diff
@@ -1295,6 +1295,14 @@ pub(crate) mod builtin {
     /// first macro invocation leading up to the invocation of the `file!`
     /// macro.
     ///
+    /// The file name is derived from the root module's source path passed to the Rust compiler
+    /// (e.g. `./src/lib.rs`)
+    /// end the sequence the compiler takes to get from root module to the module containing `file!`.
+    /// (e.g. inside `mod foo;` in `./src/lib.rs`, `file!` would be `./src/foo.rs`),
```

Comment:
...or `./src/foo/mod.rs`.  Perhaps:

```suggestion
    /// (e.g. if the root file of the crate is `./src/lib.rs`, and within that
    /// crate there is a module `foo` whose source file is `./src/foo.rs`, then
    /// `file!` called within will return `./src/foo.rs`),
```

Reply:
Agreed overall, though the full stop ahead of the parenthetical seems an error, as the sentence does continue after the parenthetical.  Probably it'd be better to keep the sentence together and state the "For example" after it.

Reply:
I pulled the example out into a sentence at the end so that the longer example doesn't break the flow of the explanation.